akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -832,15 +845,41 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
     }
   }
 
+  /**
+   * State transition to ShuttingDown. The current node sets its self as ShuttingDown
+   * Every other node, as soon as it sees any node as shutting down, sets its self to
+   * shutting down.
+   */
+  def startShutdown(address: Address): Unit = {
+    val memberToShutdown: Option[Member] = latestGossip.members.find(_.address == address)
+    memberToShutdown match {
+      case None =>
+        logWarning("Address {} does not exist. Shutdown not happening.", address)
+      case Some(member) =>
+        val newMembers = latestGossip.members - member + member.copy(status = PreparingForShutdown)
```

Comment:
Couldn't we change the status of all members right away? doesn't have to be done by each member itself. I think that would result in quicker dissemination, convergence.

Reply:
👍 i did consider that. Not sure why i went with this option

Reply:
Updated to change all members in one go now